dev/archery/archery/linking.py

```diff
@@ -61,9 +63,83 @@ def list_dependency_names(self):
             names.append(name)
         return names
 
+    def _remove_weak_symbols(self, symbol_info):
+        return [line for line in symbol_info if not line.endswith(" w")]
+
+    def _capture_symbols(self, remove_symbol_versions, symbol_info):
+        if remove_symbol_versions:
+            symbol_info = [line.split('@')[0].strip() for line in symbol_info]
+        return symbol_info
+
+    def list_symbols_for_dependency(self, dependency, remove_symbol_versions=False):
+        if dependency == 'linux-vdso.so.1':
+            # this is a virtual library, thus symbols cannot be listed
+            return []
+        result = _nm.run('-D', '-P', dependency, stdout=subprocess.PIPE)
+        lines = result.stdout.decode('utf-8').splitlines()
+        lines = self._capture_symbols(remove_symbol_versions, lines)
```

Comment:
Please replace this with something like:
```suggestion
        if remove_symbol_versions:
            lines = self._remove_symbol_versions(lines)
```